modules/Bio/EnsEMBL/Analysis/Hive/RunnableDB/HivePseudogenes.pm

```diff
@@ -171,16 +180,27 @@ sub run {
   my ($self) = @_;
 
   my $output_path = $self->param('output_path');
-  unless(open(OUT,">".$output_path."/all_multi_exon_genes.fasta")) {
-    $self->throw("Could not create all_multi_exon_genes.fasta for writing. Path used:\n".$output_path."/all_multi_exon_genes.fasta")
+  my $single_multi_file = $self->param('single_multi_file');
+  my $iid_slice_name = $self->param('iid');
+  my $all_multi_exons_filename = 'all_multi_exon_genes.fasta';
+
+  if ($single_multi_file) {
+
+    if ($iid_slice_name) {
+      $all_multi_exons_filename = 'all_multi_exon_genes_'.$iid_slice_name.'.fasta';
+    }
+    
+    unless (open(OUT,">".$output_path."/".$all_multi_exons_filename)) {
+      $self->throw("Could not create ".$all_multi_exons_filename." for writing. Path used:\n".$output_path."/".$all_multi_exons_filename)
+    }
```

Comment:
The open should be out of the if statement otherwise the file will not be open if single_multi_file is false.
I may have missed something
The if statement with iid_slice_name can be merged with the if above unless there should be an else in the case iid_slice_name is "false"